envisage/tests/test_extension_point.py

```diff
@@ -120,21 +122,56 @@ def test_mutate_extension_point_no_effect(self):
         registry.add_extension_point(self._create_extension_point("my.ep"))
 
         # Set the extensions.
-        registry.set_extensions("my.ep", [1, 2, 3])
+        registry.set_extensions("my.ep", [1, 2, 3, 0, 5])
 
         # Declare a class that consumes the extension.
         class Foo(TestBase):
             x = ExtensionPoint(List(Int), id="my.ep")
 
         # when
         f = Foo()
-        f.x.append(42)
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.append(42)
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.clear()
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.extend((100, 101))
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.insert(0, 1)
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.pop()
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.remove(1)
+
+        with self.assertWarns(RuntimeWarning):
+            f.x[0] = 99
+
+        with self.assertWarns(RuntimeWarning):
+            f.x *= 99
+
+        with self.assertWarns(RuntimeWarning):
+            f.x += [9]
+
+        with self.assertWarns(RuntimeWarning):
+            del f.x[0:2]
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.reverse()
+
+        with self.assertWarns(RuntimeWarning):
+            f.x.sort()
 
         # then
         # The registry is not changed, and the extension point is still the
         # same as before
-        self.assertEqual(registry.get_extensions("my.ep"), [1, 2, 3])
-        self.assertEqual(f.x, [1, 2, 3])
+        self.assertEqual(registry.get_extensions("my.ep"), [1, 2, 3, 0, 5])
+        self.assertEqual(f.x.copy(), [1, 2, 3, 0, 5])
```

Comment:
The change to `f.x.copy()` makes the test more strict and meaningful because it falls back to getting the actual values on the list, otherwise the `__eq__` logic is used for this comparison and we would just be comparing the values on the registry.